rdrf/rdrf/filestorage.py

```diff
@@ -1,39 +1,76 @@
-# Attempt to clean up the interaction with GridFS
 import logging
+import re
+import gridfs
+from bson.objectid import ObjectId
+from .models import Registry, CDEFile
```

Comment:
The multisection index field would be tricky because they might need to be updated if a section is deleted.

I was thinking the same thing, that the `CDEFile` is implicitly linked to the mongo document. They might come in handy. I would prefer to keep the references for now and ignore them when necessary.